writer.go

```diff
@@ -704,8 +712,29 @@ func (w *writer) write(conn *Conn, batch []Message, resch [](chan<- error)) (ret
 
 	t0 := time.Now()
 	conn.SetWriteDeadline(time.Now().Add(w.writeTimeout))
-	if _, err = conn.WriteCompressedMessages(w.codec, batch...); err != nil {
-		w.stats.errors.observe(1)
+	attempts := 0
+	for w.retries > attempts {
```

Comment:
If `w.retries` is `0` you wont enter this for loop, shouldn't this be `for w.retries >= attempts`?  I guess this asks the question of whether we're defining retries as the number of retries or the number or total attempted requests?

Reply:
yeah, I see that. 
retries as the number of retires. 
Then I should start attempts at `-1` or change this logic completely.

Reply:
I changed, the logic. PR incoming.